### PR TITLE
 Detect and terminate stopped nodes

### PR DESCRIPTION
**Pull request by twiest**
- [x] Revert commit of initial attempt at removing stopped instances (aws shutdownBehavior doesn't work in all cases).
- [x] Remove stopped nodes when new nodes are created for the same machine.
- [x] Explicitly do NOT remove stopped master instances.
- [x] Fix missing clusterid in MachineSetSpec in aws-actuator-test/main.go
- [x] Fix go linting issues in aws-actuator-test/main.go
- [x] Fix cli help messages in aws-actuator-test/main.go

### CODE REVIEW

**Comment by twiest** [comment]
@csrwng Would you mind taking a look at this please?

File: pkg/clusterapi/aws/actuator.go [hunk 1]
```diff
@@ -115,9 +113,30 @@ func (a *Actuator) Create(cluster *clusterv1.Cluster, machine *clusterv1.Machine
...
+
+	return TerminateInstances(client, instances, mLog)
+}
+
 // CreateMachine starts a new AWS instance as described by the cluster and machine resources
 func (a *Actuator) CreateMachine(cluster *clusterv1.Cluster, machine *clusterv1.Machine) (*ec2.Instance, error) {
 	mLog := clustoplog.WithMachine(a.logger, machine)
+
+	isMaster := false
+	if controller.MachineHasRole(machine, capicommon.MasterRole) {
```
Comment (csrwng) [inline pkg/clusterapi/aws/actuator.go]: Just a nit, you can collapse a few lines by just doing
```go
isMaster := controller.MachineHasRole(machine, capicommon.MasterRole)
```

**Comment by twiest** [comment]
@csrwng thanks for the review and for pointing out that optimization. I believe that I've addressed your comment. PTAL.

**Pull request merged**